File: day5/exercises-solutions/hello-barrier.f90
PROGRAM hello_barrier
  USE mpi
  IMPLICIT NONE
  INTEGER :: i, size, rank, ierr

  CALL MPI_INIT(ierr)
  CALL MPI_COMM_SIZE(MPI_COMM_WORLD,size,ierr)
  CALL MPI_COMM_RANK(MPI_COMM_WORLD,rank,ierr)

  DO i=1,size
      CALL MPI_BARRIER(MPI_COMM_WORLD,ierr)
      IF (rank == i-1) THEN
          PRINT*,"Hello, World! I am rank ",rank," out of a total of",size," tasks"
      END IF
  END DO
  CALL MPI_FINALIZE(ierr)
END PROGRAM hello_barrier

File: day5/exercises-solutions/hello-barrier.c
#include <mpi.h>
#include <stdio.h>

int main(int argc, char **argv)
{
    int i,size,rank;
    MPI_Init(&argc, &argv);
    MPI_Comm_size(MPI_COMM_WORLD,&size);
    MPI_Comm_rank(MPI_COMM_WORLD,&rank);

    for (i=0; i < size; ++i) {
        MPI_Barrier(MPI_COMM_WORLD);
        if (rank == i)
            printf("Hello, World! I am rank %d "
                   "out of a total of %d tasks\n", rank, size);
    }
    MPI_Finalize();
    return 0;
}
